From a3833d179584cc723d38c07ed715446f314ce348 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Thu, 11 Jul 2019 13:01:41 +0200
Subject: [PATCH] Rework how the fdt command builds.

Trying to avoid all variants of:
cat syminfo.lst | sort | gawk -f ../../grub-core/genmoddep.awk > moddep.lst || (rm -f moddep.lst; exit 1)
grub_fdt_install in linux is not defined
grub_fdt_load in linux is not defined
grub_fdt_unload in linux is not defined
grub_fdt_install in xen_boot is not defined
grub_fdt_load in xen_boot is not defined
grub_fdt_unload in xen_boot is not defined

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 grub/grub-core/Makefile.am       | 1 +
 grub/grub-core/Makefile.core.def | 7 ++++---
 grub/grub-core/lib/fdt.c         | 2 --
 grub/grub-core/loader/efi/fdt.c  | 2 ++
 grub/include/grub/fdt.h          | 4 ++++
 5 files changed, 11 insertions(+), 5 deletions(-)

diff --git a/grub/grub-core/Makefile.am b/grub/grub-core/Makefile.am
index 3ea8e7ff4..d61236924 100644
--- a/grub/grub-core/Makefile.am
+++ b/grub/grub-core/Makefile.am
@@ -74,6 +74,7 @@ KERNEL_HEADER_FILES += $(top_srcdir)/include/grub/dl.h
 KERNEL_HEADER_FILES += $(top_srcdir)/include/grub/env.h
 KERNEL_HEADER_FILES += $(top_srcdir)/include/grub/env_private.h
 KERNEL_HEADER_FILES += $(top_srcdir)/include/grub/err.h
+KERNEL_HEADER_FILES += $(top_srcdir)/include/grub/fdt.h
 KERNEL_HEADER_FILES += $(top_srcdir)/include/grub/file.h
 KERNEL_HEADER_FILES += $(top_srcdir)/include/grub/fs.h
 KERNEL_HEADER_FILES += $(top_srcdir)/include/grub/i18n.h
diff --git a/grub/grub-core/Makefile.core.def b/grub/grub-core/Makefile.core.def
index 474a63e68..c0cc8a4ec 100644
--- a/grub/grub-core/Makefile.core.def
+++ b/grub/grub-core/Makefile.core.def
@@ -169,7 +169,6 @@ kernel = {
   arm_coreboot = kern/arm/coreboot/init.c;
   arm_coreboot = kern/arm/coreboot/timer.c;
   arm_coreboot = kern/arm/coreboot/coreboot.S;
-  arm_coreboot = lib/fdt.c;
   arm_coreboot = bus/fdt.c;
   arm_coreboot = term/ps2.c;
   arm_coreboot = term/arm/pl050.c;
@@ -340,6 +339,8 @@ kernel = {
   riscv64 = kern/riscv/cache_flush.S;
   riscv64 = kern/riscv/dl.c;
 
+  fdt = lib/fdt.c;
+
   emu = disk/host.c;
   emu = kern/emu/cache_s.S;
   emu = kern/emu/hostdisk.c;
@@ -1784,7 +1784,6 @@ module = {
   riscv32 = loader/riscv/linux.c;
   riscv64 = loader/riscv/linux.c;
   emu = loader/emu/linux.c;
-  fdt = lib/fdt.c;
 
   common = loader/linux.c;
   common = lib/cmdline.c;
@@ -1794,7 +1796,6 @@ module = {
 module = {
   name = fdt;
   efi = loader/efi/fdt.c;
-  common = lib/fdt.c;
   enable = fdt;
 };
 
diff --git a/grub/grub-core/lib/fdt.c b/grub/grub-core/lib/fdt.c
index 0d371c563..37e04bd69 100644
--- a/grub/grub-core/lib/fdt.c
+++ b/grub/grub-core/lib/fdt.c
@@ -21,8 +21,6 @@
 #include <grub/mm.h>
 #include <grub/dl.h>
 
-GRUB_MOD_LICENSE ("GPLv3+");
-
 #define FDT_SUPPORTED_VERSION	17
 
 #define FDT_BEGIN_NODE	0x00000001
diff --git a/grub/grub-core/loader/efi/fdt.c b/grub/grub-core/loader/efi/fdt.c
index ee9c5592c..37ca407bb 100644
--- a/grub/grub-core/loader/efi/fdt.c
+++ b/grub/grub-core/loader/efi/fdt.c
@@ -26,6 +26,8 @@
 #include <grub/efi/fdtload.h>
 #include <grub/efi/memory.h>
 
+GRUB_MOD_LICENSE ("GPLv3+");
+
 static void *loaded_fdt;
 static void *fdt;
 
diff --git a/grub/include/grub/fdt.h b/grub/include/grub/fdt.h
index e609c7e41..22b7c5463 100644
--- a/grub/include/grub/fdt.h
+++ b/grub/include/grub/fdt.h
@@ -19,6 +19,8 @@
 #ifndef GRUB_FDT_HEADER
 #define GRUB_FDT_HEADER	1
 
+#if defined(__arm__) || defined(__aarch64__)
+
 #include <grub/types.h>
 #include <grub/symbol.h>
 
@@ -144,4 +146,6 @@ int EXPORT_FUNC(grub_fdt_set_prop) (void *fdt, unsigned int nodeoffset, const ch
   grub_fdt_set_prop ((fdt), (nodeoffset), "reg", reg_64, 16);  \
 })
 
+#endif /* defined(__arm__) || defined(__aarch64__) */
+
 #endif	/* ! GRUB_FDT_HEADER */
-- 
2.25.4

